src/defrag.c

```diff
@@ -770,17 +770,22 @@ void defragScanCallback(void *privdata, const dictEntry *de) {
  * or not, a false detection can cause the defragmenter to waste a lot of CPU
  * without the possibility of getting any results. */
 float getAllocatorFragmentation(size_t *out_frag_bytes) {
-    size_t resident, active, allocated;
-    zmalloc_get_allocator_info(&allocated, &active, &resident);
-    float frag_pct = ((float)active / allocated)*100 - 100;
-    size_t frag_bytes = active - allocated;
+    size_t resident, active, allocated, frag_smallbins_bytes;
+    zmalloc_get_allocator_info(&allocated, &active, &resident, &frag_smallbins_bytes);
+    serverAssert(frag_smallbins_bytes == (active - allocated)); // Just for verify, removed future
+
+    /* Calculate the fragmentation ratio as the proportion of wasted memory in small
+     * bins relative to the total allocated memory (including large bins). This approach
+     * is because when we mostly use large bins, even though the fragmentation doesn't
+     * consume a lot of memory, it will show a high percentage. */
```

Comment:
```suggestion
    /* Calculate the fragmentation ratio as the proportion of wasted memory in small
     * bins (which are defraggable) relative to the total allocated memory (including large bins).
     * This is because otherwise, if most of the memory usage is large bins, we may show high percentage,
     * despite the fact it's not a lot of memory for the user. */
```